packages/dds/merge-tree/src/client.ts

```diff
@@ -996,18 +997,46 @@ export class Client {
         }
     }
 
-    getContainingSegment<T extends ISegment>(pos: number, op?: ISequencedDocumentMessage) {
-        let seq: number;
-        let clientId: number;
+    private getRefSeqAndClientId(op?: ISequencedDocumentMessage) {
         if (op) {
-            clientId = this.getOrAddShortClientId(op.clientId);
-            seq = op.referenceSequenceNumber;
+            return { refSeq: op.referenceSequenceNumber, clientId: this.getOrAddShortClientId(op.clientId) };
         } else {
             const segWindow = this.mergeTree.getCollabWindow();
-            seq = segWindow.currentSeq;
-            clientId = segWindow.clientId;
+            return { refSeq: segWindow.currentSeq, clientId: segWindow.clientId };
         }
-        return this.mergeTree.getContainingSegment<T>(pos, seq, clientId);
+    }
+
+    getContainingSegment<T extends ISegment>(pos: number, op?: ISequencedDocumentMessage) {
+        const { refSeq, clientId } = this.getRefSeqAndClientId(op);
+        return this.mergeTree.getContainingSegment<T>(pos, refSeq, clientId);
+    }
+
+    createSlideOnRemoveReference(refType: ReferenceType, pos: number, pending: boolean, op?: ISequencedDocumentMessage):
+        ReferencePosition {
+        // TODO:ransomr should this assert, throw, or something else if refType is invalid?
+        // eslint-disable-next-line no-bitwise
+        assert((refType & ReferenceType.SlideOnRemove) > 0, "Reference type must be SlideOnRemove");
```

Comment:
I'd suggest throwing a `UsageError` here (from `@fluidframework/container-utils`), since this is just argument validation.